src/ws/test-webservice.c

```diff
@@ -449,7 +449,7 @@ close_client_and_stop_web_service (TestCase *test,
   timeout = g_timeout_add_seconds (20, on_timeout_fail, "closing web service");
   g_object_add_weak_pointer (G_OBJECT (test->service), (gpointer *)&test->service);
   g_object_unref (test->service);
-  while (test->service != NULL)
+  while (test->service != NULL) // check_after_deref: Null-checking "test->service" suggests that it may be null, but it has already been dereferenced on all paths leading to the check. 
```

Comment:
@allisonkarlitskaya WDYT of this report? I am not sure what it means so I just wanted to raise it with someone who understands this.

Reply:
This coverity warning is wrong. Of course test->service is not NULL *initially*, but the point is that the expected callbacks eventually unref it and thus cause it to get set *to* NULL. This happens asynchronously (well, not really, but you can think about it that way). If that were an `if`, coverity would be right, but apparnetly it doesn't understand this `while` loop and that test->service is mutated by the main loop.